lib/spack/spack/repo.py

```diff
@@ -1064,7 +1068,16 @@ def all_package_classes(self):
 
     def exists(self, pkg_name):
         """Whether a package with the supplied name exists."""
-        return pkg_name in self._pkg_checker
+        if pkg_name is None:
+            return False
+
+        # if the FastPackageChecker is already constructed, use it
+        if self._fast_package_checker:
+            return pkg_name in self._pkg_checker
+
+        # if not, check for the package.py file
+        path = self.filename_for_package_name(pkg_name)
+        return os.path.exists(path)
```

Comment:
Are there situations where these two checks would give different results? I can't think on any at the moment, but here we open for potentially different answers depending on `self._fast_package_checker`. What would be the difference between this and memoizing?

Reply:
only if you built the package checker and a package was removed before that particular spack run completed. I consider that user error, though -- the only command that really runs long enough for it to happen is `spack install`, and if you delete a `package.py` while it's installing that seems like a bigger issue.